File: evals/fixtures/fastapi-app/src/services/order.py
from src.db.repository import OrderRepository
from src.services.payment import charge

async def create_order(payload, user):
    order = await OrderRepository.create(payload, user["id"])
    result = await charge(order)
    if not result["ok"]:
        await OrderRepository.mark_failed(order["id"], result["reason"])
        raise PaymentError(result["reason"])
    return order

async def cancel_order(order_id):
    await OrderRepository.update_status(order_id, "cancelled")

def _validate(payload):
    return payload is not None

File: evals/fixtures/express-app/src/services/order.ts
import { OrderRepository } from '../repositories/order'
import { chargePayment } from './payment'
import { logger } from '../utils/logger'

export async function createOrder(input: OrderInput, user: User): Promise<Order> {
  const order = await OrderRepository.create({ ...input, userId: user.id })
  const charge = await chargePayment(order)
  if (!charge.ok) {
    await OrderRepository.markFailed(order.id, charge.reason)
    logger.warn('charge failed', { orderId: order.id })
    throw new PaymentError(charge.reason)
  }
  return order
}

export async function cancelOrder(id: string): Promise<void> {
  await OrderRepository.updateStatus(id, 'cancelled')
}
